bundle/artifacts/whl/build.go

```diff
@@ -0,0 +1,61 @@
+package whl
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs"
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/cli/python"
+)
+
+type build struct {
+	name string
+}
+
+func Build(name string) bundle.Mutator {
+	return &build{
+		name: name,
+	}
+}
+
+func (m *build) Name() string {
+	return fmt.Sprintf("artifacts.whl.Build(%s)", m.name)
+}
+
+func (m *build) Apply(ctx context.Context, b *bundle.Bundle) error {
+	artifact, ok := b.Config.Artifacts[m.name]
+	if !ok {
+		return fmt.Errorf("artifact doesn't exist: %s", m.name)
+	}
+
+	if artifact.BuildCommand == "" {
+		//TODO: infer build step if not provided
```

Comment:
Inferring this should happen as a mutator prior to this one, so that it can be observable.